pkg/convert/jfr/parser.go

```diff
@@ -14,155 +14,224 @@ import (
+			pi.AggregationType = metadata.SumAggregationType
+			if putErr := s.Put(ctx, pi); putErr != nil {
+				err = multierror.Append(err, putErr)
+			}
+			labels["__name__"] = prefix + ".alloc_in_new_tlab_bytes"
+			pi.Key = segment.NewKey(labels)
+			pi.Val = inTLABBytes
+			pi.Units = metadata.BytesUnits
+			pi.AggregationType = metadata.SumAggregationType
+			if putErr := s.Put(ctx, pi); putErr != nil {
+				err = multierror.Append(err, putErr)
+			}
+			labels["__name__"] = prefix + ".alloc_outside_tlab_objects"
+			pi.Key = segment.NewKey(labels)
+			pi.Val = outTLABObjects
+			pi.Units = metadata.ObjectsUnits
+			pi.AggregationType = metadata.SumAggregationType
+			if putErr := s.Put(ctx, pi); putErr != nil {
+				err = multierror.Append(err, putErr)
+			}
+			labels["__name__"] = prefix + ".alloc_outside_tlab_bytes"
+			pi.Key = segment.NewKey(labels)
+			pi.Val = outTLABBytes
+			pi.Units = metadata.BytesUnits
+			pi.AggregationType = metadata.SumAggregationType
+			if putErr := s.Put(ctx, pi); putErr != nil {
+				err = multierror.Append(err, putErr)
+			}
 		}
-		labels["__name__"] = prefix + ".alloc_outside_tlab_bytes"
-		pi.Key = segment.NewKey(labels)
-		pi.Val = outTLABBytes
-		pi.Units = metadata.BytesUnits
-		pi.AggregationType = metadata.SumAggregationType
-		if putErr := s.Put(ctx, pi); putErr != nil {
-			err = multierror.Append(err, putErr)
+		if lock != "" {
+			labels["__name__"] = prefix + ".lock_count"
+			pi.Key = segment.NewKey(labels)
+			pi.Val = lockSamples
+			pi.Units = metadata.LockSamplesUnits
+			pi.AggregationType = metadata.SumAggregationType
+			if putErr := s.Put(ctx, pi); putErr != nil {
+				err = multierror.Append(err, putErr)
+			}
+			labels["__name__"] = prefix + ".lock_duration"
+			pi.Key = segment.NewKey(labels)
+			pi.Val = lockDuration
+			pi.Units = metadata.LockNanosecondsUnits
+			pi.AggregationType = metadata.SumAggregationType
+			if putErr := s.Put(ctx, pi); putErr != nil {
+				err = multierror.Append(err, putErr)
+			}
 		}
 	}
-	if lock != "" {
-		labels["__name__"] = prefix + ".lock_count"
-		pi.Key = segment.NewKey(labels)
-		pi.Val = lockSamples
-		pi.Units = metadata.LockSamplesUnits
-		pi.AggregationType = metadata.SumAggregationType
-		if putErr := s.Put(ctx, pi); putErr != nil {
-			err = multierror.Append(err, putErr)
-		}
-		labels["__name__"] = prefix + ".lock_duration"
-		pi.Key = segment.NewKey(labels)
-		pi.Val = lockDuration
-		pi.Units = metadata.LockNanosecondsUnits
-		pi.AggregationType = metadata.SumAggregationType
-		if putErr := s.Put(ctx, pi); putErr != nil {
-			err = multierror.Append(err, putErr)
+
+	return err
+}
+
+func groupEventsByContextId(events []parser.Parseable) map[int64][]parser.Parseable {
+	res := make(map[int64][]parser.Parseable)
+	for _, e := range events {
+		switch e.(type) {
+		case *parser.ExecutionSample:
+			es := e.(*parser.ExecutionSample)
+			contextId := es.ContextId
+			res[contextId] = append(res[contextId], e)
```

Comment:
nit: these can just be `res[es.ContextId`] = ...` (no need for an extra variable imo)

I don't have a strong opinion on this though.

Reply:
inlined